File: BotNet/client/pristine/python/AI.py
#-*-python-*-
from BaseAI import BaseAI
from GameObject import *

class AI(BaseAI):
  """The class implementing gameplay logic."""
  @staticmethod
  def username():
    return "Shell AI"

  @staticmethod
  def password():
    return "password"

  def init(self):
    pass

  def end(self):
    pass

  def run(self):
    # loop through all of the bases
    for base in self.bases:
      # check if you own that base
      if base.getOwner() == self.playerID():
        # check to see if you have enough cycles to spawn a level 0 virus
        if self.baseCost() <= self.players[self.playerID()].getCycles():
          # spawn a level 0 virus at that base
          base.spawn(0)

    # loop through all of the viruses
    for virus in self.viruses:
      # if you own that virus
      if virus.getOwner() == self.playerID():
        # if the tile you want to move to is NOT a wall
        if self.getTileAtLocation(virus.getX()+1, virus.getY()).getOwner() != 3:
          # move the virus
          virus.move(virus.getX()+1, virus.getY())
    
    # End your turn
    return True;

  def __init__(self, conn):
      BaseAI.__init__(self, conn)

File: BotNet/client/pristine/c/AI.cpp
#include "AI.h"
#include "util.h"

#include "shell.h"

AI::AI(Connection* conn) : BaseAI(conn) {}

const char* AI::username()
{
  return "mst00120";
}

const char* AI::password()
{
  return "aegoaf5Y";
}

//This function is run once, before your first turn.
void AI::init(){}

//This function is called each time it is your turn.
//Return true to end your turn, return false to ask the server for updated information.
bool AI::run()
{
  Map map(this);
  build_shells(this,map);

  // loop through all of the bases
  for(int i=0;i<bases.size();i++)
  {
    // check if you own that base
    if(bases[i].owner() == playerID())
    {
      // check to see if you have enough cycles to spawn a level 0 virus
      if(baseCost() <= players[playerID()].cycles())
      {
        // spawn a level 0 virus at that base
        bases[i].spawn(0);
      }
    }
  }

  // // loop through all of the viruses
  // for(int i=0;i<viruses.size();i++)
  // {
  //   // if you own that virus
  //   if(viruses[i].owner() == playerID())
  //   {
  //     // if the tile you want to move to is NOT a wall
  //     if(getTileAtLocation(viruses[i].x()+1, viruses[i].y()).owner() != 3)
  //     {
  //       // move the virus
  //       viruses[i].move(viruses[i].x()+1, viruses[i].y());
  //     }
  //   }
  // }
  // End your turn
  return true;
}

//This function is run once, after your last turn.
void AI::end(){}
